lib/install/operation.go

```diff
@@ -165,10 +157,21 @@ func (i *Installer) completeOperation(operation ops.SiteOperation, status dispat
 // wait blocks until either the context has been cancelled or the wizard process
 // exits with an error.
 func (i *Installer) wait() error {
-	i.stopStoppers(i.ctx)
+	i.runStoppers(i.ctx)
 	return trace.Wrap(i.config.Process.Wait())
 }
 
+func (i *Installer) registerExitHandlersForAgents(op ops.SiteOperation) {
+	i.addAborter(signals.StopperFunc(func(ctx context.Context) error {
+		i.WithField("operation", op.ID).Info("Aborting agent service.")
+		return trace.Wrap(i.config.Process.AgentService().AbortAgents(ctx, op.Key()))
+	}))
+	i.addStopper(signals.StopperFunc(func(ctx context.Context) error {
+		i.WithField("operation", op.ID).Info("Stopping agent service.")
```

Comment:
Agents are restarted each time installer restarts.
Agents are aborted only if the installer has been interrupted.